httpclient/src/test/java/feign/httpclient/ApacheHttpClientTest.java

```diff
@@ -54,19 +56,44 @@ public void parsesRequestAndResponse() throws IOException, InterruptedException
         .target(TestInterface.class, "http://localhost:" + server.getPort());
 
     Response response = api.post("foo");
+    try {
+
+      assertThat(response.status()).isEqualTo(200);
+      assertThat(response.reason()).isEqualTo("OK");
+      assertThat(response.headers())
+              .containsEntry("Content-Length", asList("3"))
+              .containsEntry("Foo", asList("Bar"));
+      assertThat(response.body().length()).isEqualTo(3);
+      assertThat(response.body().asInputStream())
+              .hasContentEqualTo(new ByteArrayInputStream("foo".getBytes(UTF_8)));
+
+      assertThat(server.takeRequest()).hasMethod("POST")
+              .hasPath("/?foo=bar&foo=baz&qux=")
+              .hasHeaders("Foo: Bar", "Foo: Baz", "Qux: ", "Accept: */*", "Content-Length: 3")
+              .hasBody("foo");
+    } finally {
+      response.close();
+    }
+  }
+
+  @Test
+  public void parsesResponseMissingLength() throws IOException, InterruptedException {
+    server.enqueue(new MockResponse().setChunkedBody("foo", 1));
 
-    assertThat(response.status()).isEqualTo(200);
-    assertThat(response.reason()).isEqualTo("OK");
-    assertThat(response.headers())
-        .containsEntry("Content-Length", asList("3"))
-        .containsEntry("Foo", asList("Bar"));
-    assertThat(response.body().asInputStream())
-        .hasContentEqualTo(new ByteArrayInputStream("foo".getBytes(UTF_8)));
-
-    assertThat(server.takeRequest()).hasMethod("POST")
-        .hasPath("/?foo=bar&foo=baz&qux=")
-        .hasHeaders("Foo: Bar", "Foo: Baz", "Qux: ", "Accept: */*", "Content-Length: 3")
-        .hasBody("foo");
+    TestInterface api = Feign.builder()
+            .client(new ApacheHttpClient())
+            .target(TestInterface.class, "http://localhost:" + server.getPort());
+
+    Response response = api.post("testing");
+    try {
+      assertThat(response.status()).isEqualTo(200);
```

Comment:
too bad we don't have a feign Response assertj class, right? this would look much nicer chained. (ps if you feel like it, make one, else no biggie)